File: python-starter/hello.py
"""
hello.py - Basic Python Starter

This is your first Python program! It demonstrates:
- Print statements
- Variables
- User input
- String formatting
"""


def main():
    """Main function - entry point of the program"""
    # Print a simple message
    print("Hello, World!")
    
    # Get user input
    name = input("What's your name? ")
    
    # Format and print a personalized greeting
    print(f"Hello, {name}!")
    
    # Try different string formatting methods
    print("Nice to meet you, {}!".format(name))
    print("Welcome to Python programming, " + name + "!")


if __name__ == "__main__":
    main()

File: c-starter/hello.c
/**
 * hello.c - Basic C Starter
 * 
 * This is your first C program! It demonstrates:
 * - Main function
 * - Printf statements
 * - Variables
 * - User input with scanf
 */

#include <stdio.h>
#include <string.h>

int main(void)
{
    // Print a simple message
    printf("Hello, World!\n");
    
    // Declare a variable
    char name[50];
    
    // Get user input
    printf("What's your name? ");
    scanf("%49s", name);  // Read up to 49 characters (leave room for null terminator)
    
    // Print a personalized greeting
    printf("Hello, %s!\n", name);
    printf("Nice to meet you, %s!\n", name);
    
    return 0;
}
